packages/typescript-estree/src/ts-estree/ts-estree.ts

```diff
@@ -52,20 +52,80 @@ export interface BaseNode {
  * They are not included in the `Node` union below on purpose because they
  * are not ever included as part of the standard AST tree.
  */
-
-export interface Token extends BaseNode {
-  type: AST_TOKEN_TYPES;
+interface BaseToken extends BaseNode {
   value: string;
-  regex?: {
+}
+
+export interface BooleanToken extends BaseToken {
+  type: AST_TOKEN_TYPES.Boolean;
+}
```

Comment:
question - do we want to further specify this to this?
```ts
export interface BooleanToken extends BaseToken {
  type: AST_TOKEN_TYPES.Boolean;
  value: 'true' | 'false';
}
```

Reply:
ideally yes, but for now i did only broad types as first step